core/src/main/scala/com.mediative.amadou/DateInterval.scala

```diff
@@ -72,14 +69,15 @@ case class DateInterval(
   override def compare(that: DateInterval) = from.compareTo(that.from)
 
   override def toString =
-    interval.defaultFormatter.format(from) + toOpt.fold("")(":" + _.toString)
+    interval.defaultFormatter.format(asDate) + toOpt.fold("")(":" + _.toString)
 
   def format(dateFormat: String): String = {
     val formatter = interval.formatter(dateFormat)
-    formatter.format(from) + toOpt.fold("")(":" + _.toString)
+    formatter.format(asDate) + toOpt.fold("")(":" + _.toString)
   }
 
-  def toTimestamp: Timestamp = new Timestamp(from.getTime)
+  def asDate = new Date(from)
```

Comment:
The mutable `Date` is still public, did you change your mind about not exposing it?
^